examples/seg/deeplabv3/README.md

```diff
@@ -0,0 +1,176 @@
+
+3. Prepare dataset
+
+- Download Pascal VOC 2012 dataset,  [VOC2012](http://host.robots.ox.ac.uk/pascal/VOC/voc2012/) and Semantic Boundaries Dataset, [SBD](https://www2.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/semantic_contours/benchmark.tgz).
+
+- Prepare training and test data list files with the path to image and annotation pairs. You could simply run `python examples/seg/deeplabv3/build_dataset/get_dataset_lst.py --data_root=[PATH TO DATA]` to generate the list files. The lines in list file should be like as follows:
+
+  ```
+  /[PATH..]/JPEGImages/2007_000032.jpg /[PATH..]/SegmentationClassGray/2007_000032.png
+  /[PATH..]/JPEGImages/2007_000039.jpg /[PATH..]/SegmentationClassGray/2007_000039.png
+  /[PATH..]/JPEGImages/2007_000063.jpg /[PATH..]/SegmentationClassGray/2007_000063.png
+  ......
+  ```
+
+- Convert training dataset to mindrecords by running  ``bulid_seg_data.py `` script.
+
+  ```shell
+  python examples/seg/deeplabv3/bulid_dataset/bulid_seg_data.py \
+  		--data_root=[root path of training data] \
+  		--data_lst=[path of data list file prepared above] \
+  		--dst_path=[path to save mindrecords] \
+  		--num_shards=8 \
+  		--shuffle=True
+  ```
+
+* In accord with paper, we train on *trainaug* dataset (voc train + SBD) and evaluate on *voc val* dataset.
+
+4. Backbone: download pre-trained backbone from MindCV, here we use [ResNet101](https://download.mindspore.cn/toolkits/mindcv/resnet/resnet101-689c5e77.ckpt).
+
+### Train
+
+Specify `deeplabv3`  or  `deeplabv3plus` in key word `model` in the config file.
+
+It is highly recommended to use **distributed training** for this DeepLabV3 and DeepLabV3+ implementation.
+
+For distributed training using **OpenMPI's `mpirun`**, simply run
+```shell
+cd mindcv  # change directory to the root of MindCV repository
+mpirun -n [# of devices] python examples/seg/deeplabv3/train.py --config [the path to the config file]
+```
+
+For distributed training with [Ascend rank table](https://github.com/mindspore-lab/mindocr/blob/main/docs/en/tutorials/distribute_train.md#12-configure-rank_table_file-for-training), configure `ascend8p.sh` as follows
+
+```shell
+#!/bin/bash
+export DEVICE_NUM=8
+export RANK_SIZE=8
+export RANK_TABLE_FILE="./hccl_8p_01234567_127.0.0.1.json"
+
+for ((i = 0; i < ${DEVICE_NUM}; i++)); do
+   export DEVICE_ID=$i
+   export RANK_ID=$i
+   python -u examples/seg/deeplabv3/train.py --config [the path to the config file]  &> ./train_$i.log &
+done
+```
+
+and start training by running:
+```shell l
+cd mindcv  # change directory to the root of MindCV repository
+bash ascend8p.sh
+```
+
+For single-device training, simply set the parameter ``distributed`` to ``False`` in config file and run:
+```shell
+cd mindcv  # change directory to the root of MindCV repository
+python examples/seg/deeplabv3/train.py --config [the path to the config file]
+```
+
+##### The training steps are as follow:
```

Comment:
whats the difference from the above code snippet?

Reply:
according to paper, you should finetune twice, with differnet os setting, to get the final best weights.
the above code snippet shows how to start a training here.  "The training steps"  emphasizes the training protocol based on paper, that you should first use os=16 and then os=8.